File: c/gpt/p121_nr_ver1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) {
    const size_t n = 10;
    char *chunk1 = (char *)calloc(n + 1, 1);
    char *chunk2 = (char *)calloc(n + 1, 1);

    if (!chunk1 || !chunk2) {
        free(chunk1);
        free(chunk2);
        return 1;
    }

    memcpy(chunk1, "ABCDEFGHIJ", n);
    memcpy(chunk2, chunk1, n);
    chunk2[n] = '\0';

    printf("%s\n", chunk2);

    free(chunk1);
    chunk1 = NULL;

    printf("%s\n", chunk2);

    free(chunk2);
    chunk2 = NULL;

    return 0;
}

File: java/gpt/p121_nr_ver1.java
import java.nio.charset.StandardCharsets;
import java.util.Arrays;

public class p121_nr_ver1 {
    public static void main(String[] args) {
        final int n = 10;

        byte[] chunk1 = new byte[n + 1];
        byte[] chunk2 = new byte[n + 1];

        byte[] data = "ABCDEFGHIJ".getBytes(StandardCharsets.US_ASCII);
        System.arraycopy(data, 0, chunk1, 0, Math.min(n, data.length));
        System.arraycopy(chunk1, 0, chunk2, 0, n);
        chunk2[n] = 0;

        System.out.println(new String(Arrays.copyOfRange(chunk2, 0, n), StandardCharsets.US_ASCII));

        chunk1 = null;

        System.out.println(new String(Arrays.copyOfRange(chunk2, 0, n), StandardCharsets.US_ASCII));

        chunk2 = null;
    }
}